indico/web/client/js/react/components/category_navigator/viewmodel.jsx

```diff
@@ -0,0 +1,148 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import {getCategoryData, getCategorySearchResults} from './api';
+import {
+  initModel,
+  startLoading,
+  updateModelFromResponse,
+  updateModelFromError,
+  prepareForLoading,
+  updateModelFromSearch,
+  updateModelFromSearchError,
+  prepareForSearch,
+  clearSearch,
+  finishLoading,
+  resetCategoryList,
+} from './model';
+
+const loadingSpinnerDelay = 1500;
+
+export class DialogViewModel extends React.Component {
```

Comment:
Please do not add new class-based components - we solely use functions (+ react hooks) in any new code)

Reply:
In which way would it add complexity? FWIW I did not have a closer look at the rest of the PR yet, I just noticed this when skimming over it.

Reply:
> In which way would it add complexity? 

I've fixed the PR already, but let me add the $0.02. I've done some research last night, and this whole point is kinda moot cause React team are apparently deprecating class components anyway. We can have a chat about this elsewhere if you would like to know what my stance is. :)